File: spring-cloud-starter/src/main/java/com/jmsoftware/maf/springcloudstarter/property/MafProjectProperties.java
package com.jmsoftware.maf.springcloudstarter.property;

import lombok.Data;
import lombok.extern.slf4j.Slf4j;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.boot.info.BuildProperties;
import org.springframework.validation.annotation.Validated;

import javax.annotation.PostConstruct;
import javax.validation.constraints.NotBlank;

/**
 * <h1>MafProjectProperties</h1>
 * <p>
 * M&F project property, containing the basic constants of the project.
 *
 * @author Johnny Miller (锺俊), email: johnnysviva@outlook.com, date: 12/29/2020 12:57 PM
 */
@Data
@Slf4j
@Validated
@ConfigurationProperties(prefix = MafProjectProperties.PREFIX)
public class MafProjectProperties {
    public static final String PREFIX = "maf.project-properties";
    private final BuildProperties buildProperties;
    /**
     * The Base package.
     */
    @NotBlank
    private String basePackage;
    /**
     * The Context path.
     */
    private String contextPath;
    /**
     * The Group id.
     */
    @NotBlank
    private String groupId;
    /**
     * The Project parent artifact id.
     */
    @NotBlank
    private String projectParentArtifactId;
    /**
     * The Project artifact id.
     */
    @NotBlank
    private String projectArtifactId;
    /**
     * The Version.
     */
    @NotBlank
    private String version;
    /**
     * The Description.
     */
    @NotBlank
    private String description;
    /**
     * The Jdk version.
     */
    @NotBlank
    private String jdkVersion;
    /**
     * The Environment.
     */
    @NotBlank
    private String environment;
    /**
     * The Url.
     */
    private String url;
    /**
     * The Inception year.
     */
    private String inceptionYear;
    /**
     * The Organization name.
     */
    private String organizationName;
    /**
     * The Organization url.
     */
    private String organizationUrl;
    /**
     * The Issue management system.
     */
    private String issueManagementSystem;
    /**
     * The Issue management url.
     */
    private String issueManagementUrl;
    /**
     * The Developer name.
     */
    private String developerName;
    /**
     * The Developer email.
     */
    private String developerEmail;
    /**
     * The Developer url.
     */
    private String developerUrl;

    @PostConstruct
    private void postConstruct() {
        log.warn("Initial bean: '{}'", this.getClass().getSimpleName());
    }
}

File: spring-cloud-starter/src/main/kotlin/com/jmsoftware/maf/springcloudstarter/property/MafProjectProperties.kt
package com.jmsoftware.maf.springcloudstarter.property

import com.jmsoftware.maf.common.util.logger
import jakarta.annotation.PostConstruct
import jakarta.validation.constraints.NotBlank
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.context.annotation.Configuration
import org.springframework.validation.annotation.Validated

/**
 * # MafProjectProperties
 *
 * M&F project property, containing the basic constants of the project.
 *
 * @author Johnny Miller (锺俊), e-mail: johnnysviva@outlook.com, date: 4/13/22 2:54 PM
 */
@Validated
@Configuration
@Suppress("unused")
@ConfigurationProperties(prefix = MafProjectProperties.PREFIX)
class MafProjectProperties {
    companion object {
        const val PREFIX = "maf.project-properties"
        private val log = logger()
    }

    /**
     * The Base package.
     */
    lateinit var basePackage: @NotBlank String

    /**
     * The Context path.
     */
    lateinit var contextPath: String

    /**
     * The Group id.
     */
    lateinit var groupId: @NotBlank String

    /**
     * The Project parent artifact id.
     */
    lateinit var projectParentArtifactId: @NotBlank String

    /**
     * The Project artifact id.
     */
    lateinit var projectArtifactId: @NotBlank String

    /**
     * The Version.
     */
    lateinit var version: @NotBlank String

    /**
     * The Description.
     */
    lateinit var description: @NotBlank String

    /**
     * The Jdk version.
     */
    lateinit var jdkVersion: @NotBlank String

    /**
     * The Environment.
     */
    lateinit var environment: @NotBlank String

    /**
     * The Url.
     */
    lateinit var url: String

    /**
     * The Inception year.
     */
    lateinit var inceptionYear: String

    /**
     * The Organization name.
     */
    lateinit var organizationName: String

    /**
     * The Organization url.
     */
    lateinit var organizationUrl: String

    /**
     * The Issue management system.
     */
    lateinit var issueManagementSystem: String

    /**
     * The Issue management url.
     */
    lateinit var issueManagementUrl: String

    /**
     * The Developer name.
     */
    lateinit var developerName: String

    /**
     * The Developer email.
     */
    lateinit var developerEmail: String

    /**
     * The Developer url.
     */
    lateinit var developerUrl: String

    @PostConstruct
    fun postConstruct() {
        log.warn("Initial bean: `${this.javaClass.simpleName}`")
    }
}
